### PR TITLE
RIO-754: improving nested stack support for changeset

### PR DESCRIPTION
**Pull request by cintiadrrezdy**
I implemented some changes to improve the experience when using stacks with a lot of nested stacks. The AWS changeset still suboptimal, but that's the best they offer us, I guess. 

  - Documented `nestedStacks` option
  - Added `nestedStackVerboseLogs` and `expandedDetails` options to control the output. It shouldn't be a breaking change at all
  - Fix the output to not error when some substacks didn't have a changeset. 
  
  
  Is this something you'd be interested merging?

### CODE REVIEW

File: vars/createChangeSet.groovy [hunk 1]
```diff
@@ -291,20 +302,22 @@ def collectNestedStacks(changes) {
...
   def changeString = ""
   String[] headers = ['Operation', 'LogicalResourceId', 'ResourceType', 'Replace', 'Details']
   ArrayList data = []
 
   changeList.each { changes ->
-    border = "\n+" + "-".multiply(7) + "+" + "-".multiply(changes.getAt(0).stack.length() + 2) + "+"
-    changeString += "${border}\n| Stack | ${changes.getAt(0).stack} |${border}" + "\n"
-    data = []
     if (changes) {
+      border = "\n+" + "-".multiply(7) + "+" + "-".multiply(changes.getAt(0).stack.length() + 2) + "+"
```
Comment (cintiadrrezdy) [inline vars/createChangeSet.groovy:314]: If I didn't move these lines inside the 'if', I get a null pointer exception for nested stacks without changes.

**Comment by cintiadrrezdy** [comment]
Mentioning @aaronwalker  and @Guslington in case you think there's something in here you'd be interested.

**Comment by aaronwalker** [comment]
@cintiadrrezdy great thanks for the contribution. We will review this PR and provide any feedback 

//cc @Guslington

**Comment by Guslington** [comment]
@cintiadrrezdy Thanks for the awesome PR! i reckon we can get rid of the `nestedStackVerboseLogs` and `expandedDetails` flags and make them the default behavior.

**Comment by cintiadrrezdy** [comment]
That's great to hear, @Guslington ! 

I'm quite happy to remove the options, simplifies the code a lot. 
But do you want the new behaviour to only be applied if using `nestedStacks` or for all cases?

**Comment by Guslington** [comment]
Thanks @cintiadrrezdy! I would add the new behavior for all cases

**Comment by cintiadrrezdy** [comment]
@Guslington I think I did all changes. I have very few cases to test, so if you could run some other tests in your stacks, it would be ideal.

**Review by Guslington — APPROVED** [review]
@cintiadrrezdy passed our tests. I'll merge this in. Thanks for your contribution!

**Pull request merged**